Create bundler leak introduction post (#35)
fastruby/blog: _posts/2019-08-20-Introducing-bundler-leak.markdown

---
layout: post
title: Introducing bundler leak
date: 2019-08-20 12:53:28
categories: ["bundler"]
author: bronzdoc

---

> In this blog post I will introduce bundler leak - A bundler plugin to check gems affected by memory leaks in your Gemfile.

Reviewer: ```suggestion
In this blog post I will introduce `bundler-leak` -- A bundler plugin to find known memory leaks in your dependencies.
```

<!--more-->

> The [bundler leak](https://github.com/rubymem/bundler-leak) plugin is a fork of the famous [bundler audit](https://github.com/rubysec/bundler-audit) plugin.

Reviewer: ```suggestion
The [`bundler-leak`](https://github.com/rubymem/bundler-leak) plugin is a fork of the famous [`bundler-audit`](https://github.com/rubysec/bundler-audit) plugin.
```

Generally speaking, could you find and replace all of these? ^

> As bundler audit, bundler leak works in conjuction of a couple of projects. The first one is what we called the [ruby-mem-advisory-db](https://github.com/rubymem/ruby-mem-advisory-db) a text based database

Reviewer: ```suggestion
Just like `bundler-audit`, `bundler-leak` works thanks to a couple of community-driven, open source projects. The first one is called [ruby-mem-advisory-db](https://github.com/rubymem/ruby-mem-advisory-db): a text-based database
```

> consisting of gems with known memory leak issues. Bundler leak will compare gems stored in this database against your Gemfile i.e

Reviewer: ```suggestion
of gems with known memory leak issues. Bundler Leak will compare gems stored in this database against your Gemfile. For example:
```

> consisting of gems with known memory leak issues. Bundler leak will compare gems stored in this database against your Gemfile i.e

Reviewer: @bronzdoc Please make sure that you mention that this all started thanks to https://github.com/ASoftCo/leaky-gems

```
Audit a project's Gemfile.lock:

    $ bundle leak

    Name: therubyracer
    Version: 0.12.1
    URL: https://github.com/cowboyd/therubyracer/pull/336
    Title: Memory leak in WeakValueMap
    Solution: upgrade to ~> 0.12.3

    Unpatched versions found!
```

> Here is a sample project where you can toy with bundler leak

Reviewer: @bronzdoc Are we missing some details here?

> The last project is called [rubymem.com](https://github.com/ombulabs/rubymem.com) - A web application that will allow you to submit new `leaky gems` to the `ruby-mem-advisory-db`

Reviewer: ```suggestion
The second project is called [rubymem.com](https://github.com/rubymem/rubymem.com): A web application that will allow you to submit new _leaky gems_ to the `ruby-mem-advisory-db`
```

<div style="text-align: center; width: 500px;">
  <img src="/blog/assets/images/rubymem/rubymem-form.png">
</div>

## Conclusion

> We want to thank you to all who made and contributed to bundler audit, the project on which this plugin is based.

Reviewer: @bronzdoc Please include https://github.com/ASoftCo/leaky-gems and all of its contributors to our thank you list. :)

> We want to thank you to all who made and contributed to bundler audit, the project on which this plugin is based.

Reviewer: ```suggestion
We want to say thanks to all the contributors who contributed to bundler audit, it was a great inspiration for this plugin! 
```

> If you want to learn more or contribute to bundler leak check the repository and fill and issue or PR.

Reviewer: ```suggestion
If you want to learn more or contribute to bundler leak check the repository and submit an issue or PR.
```
